### ISSUE TITLE
Base components: implement date picker

### ISSUE THREAD

**Issue by sergei-maertens** [open-formulieren/open-forms#3060]
Epic: #2471

Users will need to provide certain dates, such as date of birth and desired appointment date.

:warning: This depends on #3056

https://nl-design-system.github.io/utrecht/storybook-react/index.html?path=/docs/react-component-calendar--docs looks like a prime candidate to build on top. Here we divert from the existing Formio date picker library (which we are not particularly happy or unhappy with).

**Tasks**

- [x] Implement the `src/components/forms/DateField.js` component
    - [x] there should be a "regular" text field input for the date value and clicking it opens the calendar/date picker in a modal/dialog
    - [x] This includes the field label
    - [x] This includes a possible field description
- [x] Add a section in Storybook under *Pure React Components / Forms / DateField* to document the component
- [x] add ability to specify disabled date ranges / disabled dates (due to reservations being fully booked, for example)
- [x] Add interaction tests for the story/stories

**Acceptance criteria**

- [x] Accessible implementation - the input can be filled via the label (click)
- [x] Accessible implementation - keyboard only navigation works properly
- [x] Can be styled primarily using the (NL Design System) design tokens, only absolutely essential CSS is added
- [x] Default Open Forms theme styles are consistent with the other component types / consistent visual identity

**Comment by sergei-maertens** [comment]
Marking this blocked for now due to ongoing upstream development efforts (ETA is ~May 13th)

**Comment by sergei-maertens** [comment]
The updated calendar component of Utrecht should now be available in `@utrecht/component-library-react@1.0.0-alpha.313`

**Comment by sergei-maertens** [comment]
Created open-formulieren/open-forms-sdk#433 for the next iteration with items that are out of scope for this issue.
